src/experiments.ts

```diff
@@ -0,0 +1,177 @@
+import { configstore } from "./configstore";
+import * as clc from "colorette";
+import { FirebaseError } from "./error";
+
+export interface Experiment<Name extends string> {
+  name: Name;
+  shortDescription: string;
+  fullDescription?: string;
+  public?: boolean;
+  docsUri?: string;
+  default?: boolean;
+}
+
+function experiment<Name extends string>(
+  name: Name,
+  shortDescription: string,
+  opts?: {
+    fullDescription?: string;
+    public?: boolean;
+    docsUri?: string;
+    default?: boolean;
+  }
+): Experiment<Name> {
+  return {
+    ...opts,
+    name,
+    shortDescription,
+  } as Experiment<Name>;
+}
+
+export const EXPERIMENTS = Object.freeze([
+  // meta:
+  experiment("experiments", "enables the experiments family of commands"),
+
+  // Realtime Database experiments
+  experiment("rtdbrules", "TODO"),
```

Comment:
@jsdt @lepatryk Can you provide a short description for these experiments? Should they be discoverable with the command `firebase experiments:enable`? If so, we should consider adding a full description and/or docs URL.

Reply:
Hmm, what is this. @jsdt is this the rules canarying that Ryan was working on? I'm not sure we should make them discoverable right now.

Reply:
Leaving them undiscoverable is the default. But we should probably still have a short description. I guess we can make it "for internal testing only"?